plugin/storage/es/ilm/policymanager.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package ilm
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"time"
+
+	"github.com/jaegertracing/jaeger/pkg/es"
+	"github.com/jaegertracing/jaeger/pkg/es/client"
+	"github.com/jaegertracing/jaeger/pkg/es/filter"
+)
+
+const (
+	DefaultIlmPolicy    = "jaeger-default-ilm-policy"
+	DefaultIsmPolicy    = "jaeger-default-ism-policy"
+	writeAliasSuffix    = "write"
+	readAliasSuffix     = "read"
+	rolloverIndexSuffix = "000001"
+	ilmVersionSupport   = 7
+	ilmPolicyFile       = "ilm-policy.json"
+	ismPolicyFile       = "ism-policy.json"
+)
+
+var ErrIlmNotSupported = errors.New("ILM is supported only for ES version 7+")
+
+// PolicyManager manages the environment for ILM or ISM policy
+type PolicyManager struct {
+	client                         func() es.Client
+	prefixedIndexNameWithSeparator string
+}
+
+// Init makes the jaeger ready for automatic rollover by using ILM by creating
+// initial rollover indices and read-write aliases
+func (p *PolicyManager) Init() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	err := p.init(ctx)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// NewPolicyManager creates the policy manager with appropriate version and prefixedIndexNameWithSeparator.
+// prefixedIndexNameWithSeparator is the prefix with separator. For example if index prefix is jaeger-main
+// and policy manager is called for span indices, then prefixedIndexNameWithSeparator will be jaeger-main-jaeger-span-
+func NewPolicyManager(cl func() es.Client, prefixedIndexNameWithSeparator string) *PolicyManager {
```

Comment:
why does everyone need to call New & Init? Can't we have just a single public method `MayebeCreatePolicy`?